stonesoup/types/state.py

```diff
@@ -147,6 +147,15 @@ class WeightedGaussianState(GaussianState):
     weight = Property(float, default=0, doc="Weight of the Gaussian State.")
 
 
+class TaggedWeightedGaussianState(WeightedGaussianState):
+    """Tagged Weighted Gaussian State Type
+
+    Gaussian State object with an associated weight and tag. Used as components
+    for a GaussianMixtureState.
+    """
+    tag = Property(int, default=0, doc="Unique tag of the Gaussian State.")
```

Comment:
This should default to `None`, and then in the `__init__` method, generate a UUID. (You can see a similar [example on the track type](https://github.com/dstl/Stone-Soup/blob/d3bcffd0c504cbb32b54eb3a603c5fc302a6e84d/stonesoup/types/track.py#L33))